spring-data-geode/src/main/java/org/springframework/data/gemfire/function/execution/GemfireFunctionProxyFactoryBean.java

```diff
@@ -98,8 +96,13 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
 
 	protected Object invokeFunction(Method method, Object[] args) {
 
-		return this.gemfireFunctionOperations
-			.executeAndExtract(this.methodMetadata.getMethodMetadata(method).getFunctionId(), args);
+		if(method.getDeclaringClass().isAnnotationPresent(OnServers.class)) {
```

Comment:
I am curious what happens when the user annotates his/her Function Execution with `OnMembers`?  It seems as though only considering `OnServers` is too limiting.

For now, I have changed the `GemFireFunctionOperations` invocation to simply always call `.execute(..)`.

Reply:
Now accounting for OnMember and OnServer